### PR TITLE
Add type annotations to `format.py` module.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Refs #2079

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM, thank you 👍

File: pylint/checkers/format.py [hunk 1]
```diff
@@ -251,13 +252,13 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
...
 
-    def __init__(self, linter=None):
+    def __init__(self, linter: PyLinter):
         super().__init__(linter)
-        self._lines = None
-        self._visited_lines = None
-        self._bracket_stack = [None]
+        self._lines: dict[int, str] = {}
+        self._visited_lines: dict[str | None, int | None] = {}
+        self._bracket_stack: list[str | None] = [None]
```
Comment (DanielNoord) [inline pylint/checkers/format.py:259]: Does this do anything? It seems we only assign `[None]` to it, or am I missing something obvious?

File: pylint/checkers/format.py [hunk 2]
```diff
@@ -251,13 +252,13 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
         ),
     )
 
-    def __init__(self, linter=None):
+    def __init__(self, linter: PyLinter):
```
Comment (DanielNoord) [inline pylint/checkers/format.py:255]:
```suggestion
    def __init__(self, linter: PyLinter) -> None:
```

File: pylint/checkers/format.py [hunk 3]
```diff
@@ -251,13 +252,13 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
...
         super().__init__(linter)
-        self._lines = None
-        self._visited_lines = None
-        self._bracket_stack = [None]
+        self._lines: dict[int, str] = {}
+        self._visited_lines: dict[str | None, int | None] = {}
```
Comment (DanielNoord) [inline pylint/checkers/format.py:258]: This seems incorrect? We use `.get()` with `ints` on this dictionary. Shouldn't it be `dict[int, int]`?

File: pylint/checkers/format.py [hunk 4]
```diff
@@ -466,7 +467,7 @@ def process_tokens(self, tokens: list[tokenize.TokenInfo]) -> None:
         if line_num == last_blank_line_num and line_num > 0:
             self.add_message("trailing-newlines", line=line_num)
 
-    def _check_line_ending(self, line_ending, line_num):
+    def _check_line_ending(self, line_ending: str, line_num: int) -> None:
```
Comment (DanielNoord) [inline pylint/checkers/format.py:470]: While we're at it, would you mind renaming `token` on line L396 and L414 to `string`? It what's being used to call this function and originally threw me off. The second argument of `TokenInfo` is called a `string` and not a `token`.

File: pylint/checkers/format.py [hunk 5]
```diff
@@ -524,15 +525,15 @@ def visit_default(self, node: nodes.NodeNG) -> None:
...
+        lines: list[str] = []
         for line in range(line, tolineno + 1):
             self._visited_lines[line] = 1
             try:
                 lines.append(self._lines[line].rstrip())
             except KeyError:
                 lines.append("")
 
-    def _check_multi_statement_line(self, node, line):
+    def _check_multi_statement_line(self, node: nodes.NG, line: str) -> None:
```
Comment (DanielNoord) [inline pylint/checkers/format.py:536]:
```suggestion
    def _check_multi_statement_line(self, node: nodes.NodeNG, line: str) -> None:
```

?

File: pylint/checkers/format.py [hunk 6]
```diff
@@ -645,7 +646,7 @@ def specific_splitlines(lines: str) -> list[str]:
     def check_lines(self, lines: str, lineno: int) -> None:
         """Check given lines for potential messages.
 
-        Check lines have :
+        Check lines have:
```
Comment (DanielNoord) [inline pylint/checkers/format.py:649]:
```suggestion
        Check if lines have:
```

File: pylint/checkers/format.py [hunk 7]
```diff
@@ -127,22 +128,22 @@ def _last_token_on_line_is(tokens, line_end, token):
 class TokenWrapper:
     """A wrapper for readable access to token information."""
 
-    def __init__(self, tokens):
+    def __init__(self, tokens: list[tokenize.TokenInfo]):
```
Comment (DanielNoord) [inline pylint/checkers/format.py:131]:
```suggestion
    def __init__(self, tokens: list[tokenize.TokenInfo]) -> None:
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: pylint/checkers/format.py (hunk 5 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/format.py:536]: Nice catch ! I always forget that as long as astroid do not expose type we can put anything here and mypy is happy with really wrong typing...

File: pylint/checkers/format.py (hunk 1 above)
Comment (mbyrnepr2) [reply pylint/checkers/format.py:259]: Good observation @DanielNoord. If you do `git show 480e0ad898` and view the module, there are some push operations to add values to `self._bracket_stack`, but as you say, we only assign `[None]` now. I'll remove it.

**Review by DanielNoord — APPROVED** [review]

**Comment by mbyrnepr2** [comment]
Sorry for late changes, just noticed one more improvement on the typing: using `Literal[1, 2]` for possible dict values.

**Pull request merged**